geomesa-core/src/main/scala/org/locationtech/geomesa/core/data/AccumuloDataStoreFactory.scala

```diff
@@ -252,4 +221,42 @@ object AccumuloDataStoreFactory {
       authsParam.key          -> conf.get(AUTHS),
       visibilityParam.key     -> conf.get(VISIBILITY),
       featureEncParam.key     -> conf.get(FEATURE_ENCODING))
+
+  def getAuthsProvider(): AuthorizationsProvider = {
+    // if the user specifies an auth provider to use, try to use that impl
+    val authProviderSystemProperty = Option(System.getProperty(AuthorizationsProvider.AUTH_PROVIDER_SYS_PROPERTY))
+
+    // we wrap the authorizations provider in one that will filter based on the max auths configured for this store
+    val authorizationsProvider = new FilteringAuthorizationsProvider ({
```

Comment:
This has become an unwieldy line.  Can you break it up into multiple functions and intermediate values?